examples/cryptocurrency-advanced/backend/src/lib.rs

```diff
@@ -48,8 +48,11 @@ pub const INITIAL_BALANCE: u64 = 100;
 pub struct CryptocurrencyService;
 
 impl Service for CryptocurrencyService {
-    fn state_hash(&self, data: BlockchainData<&dyn Snapshot>) -> Vec<Hash> {
-        Schema::new(data.for_executing_service()).state_hash()
+    fn initialize(&self, context: CallContext<'_>, _params: Vec<u8>) -> Result<(), ExecutionError> {
+        // Initialize indexes. Not doing this may lead to errors in HTTP API, since it relies on
+        // `wallets` indexes being initialized for returning corresponding proofs.
+        Schema::new(context.service_data());
```

Comment:
Just a thought. Maybe it worth adding a type `Schema` to the `Service` trait and have a default implementation of `Service::initialize` which'll call `Service::Schema::new(context.service_data())`?

In theory it should reduce amount of boilerplate and protect those fellows who'll forget to do it.

Reply:
This will not work because traits with associated types are not object safe. We have to create separated trait and then convert it to the `Box<dyn Service>`.

Reply:
Indeed.